### PR TITLE
feat: set up env for test & add basic unit test example

### PR DESCRIPTION
**Pull request by 0xSulpiride**
## Description

- added basic unit test for ReputationService, UserOpValidationService and EthModule
- tests are run by "npm run test"
- Tests use Anvil so in order to run them you must have Anvil installed - https://book.getfoundry.sh/getting-started/installation

## Types of changes
What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Further comments (optional)

-

### CODE REVIEW

**Comment by 0xSulpiride** [comment]
Apparently viem library does not carry it's own evm for tests, it makes use of anvil or hardhat. We can use that with ethers.js as well, so I went on using Ethers in tests.
To run tests you must have Anvil installed: https://book.getfoundry.sh/getting-started/installation

**Comment by 0xSulpiride** [comment]
I'll add some more unit tests for each module and service

File: package-lock.json
Comment (0xSulpiride) [inline package-lock.json:1]: please delete this file. we only use yarn.lock

File: package.json [hunk 1]
```diff
@@ -14,11 +14,13 @@
     "fix-lint": "eslint --ext .ts --fix packages/*/src/",
     "test": "lerna run test --concurrency 1",
     "test:unit": "lerna run test:unit --no-bail --concurrency 1",
+    "coverage": "vitest run --coverage",
     "check-readme": "lerna run check-readme",
     "release:publish": "lerna publish from-package --yes --no-verify-access",
     "release:docker": "exit 0"
   },
   "dependencies": {
+    "@flashbots/ethers-provider-bundle": "^0.6.2",
```
Comment (0xSulpiride) [inline package.json:23]: >"@flashbots/ethers-provider-bundle": "^0.6.2",

delete this dependency from here, we only use it in `executors` package

File: yarn.lock
Comment (0xSulpiride) [inline yarn.lock:1]: also pls revert your changes to `yarn.lock` file

**Review by 0xSulpiride — COMMENTED** [review]
Great job on tests @Jineshdarjee! I have some minor comments

File: package-lock.json
Comment (Jineshdarjee) [reply package-lock.json:1]: Removed the package-lock.json file

File: package.json (hunk 1 above)
Comment (Jineshdarjee) [reply package.json:23]: Removed the mentioned dependency from the package.json file

File: yarn.lock
Comment (Jineshdarjee) [reply yarn.lock:1]: Created new yarn.lock file

Comment (0xSulpiride) [reply yarn.lock:1]: @Jineshdarjee pls revert it completely. We need yarn.lock that was created on Linux, the one you submitted is on Windows

Comment (Jineshdarjee) [reply yarn.lock:1]: @0xSulpiride Reverted the yarn.lock file

**Review by ch4r10t33r — APPROVED** [review]

**Pull request merged**